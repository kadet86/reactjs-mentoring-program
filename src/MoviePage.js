import React from 'react';
import Movie from './Movie';
import GenreMovieList from './GenreMovieList';

export default function MoviePage({movie, navigateBack, navigateToMovie}) {
    return (
        <section>
            <div className="top-section">
                <div>
                    <button onClick={navigateBack}>Search</button>
                    <Movie movie={movie} />
                </div>
            </div>
            <GenreMovieList genre={movie.genres[0]} navigateToMovie={navigateToMovie} />
        </section>
    );
}