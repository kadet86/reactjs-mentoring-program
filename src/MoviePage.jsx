import Router from 'next/router';
import { Button } from 'primereact/button';
import React from 'react';
import { connect } from 'react-redux';
import { getMovie, showMovie } from './actions';
import { buildSearchRoute } from './FilteredMovieList';
import GenreMovieList from './GenreMovieList';
import Movie from './Movie';
import TopSection from './TopSection';

class MoviePage extends React.PureComponent {
    componentDidMount() {
        this.fetchMovie();
    }

    componentDidUpdate(prevProps) {
        if (prevProps.router.query.id !== this.props.router.query.id) {
            document.documentElement.scrollTop = 0;
            this.fetchMovie();
        }
    }

    fetchMovie() {
        this.props.getMovie({ id: this.props.router.query.id });
    }

    navigateToSearch = () => {
        const { url, as } = buildSearchRoute(this.props);
        Router.push(url, as);
    };

    render() {
        const movie = this.props.movie || {};
        const genre = movie && movie.genres && movie.genres[0];
        return (
            <section className="movie-page">
                <TopSection>
                    <Button
                        onClick={this.navigateToSearch}
                        label="SEARCH"
                        className="p-button-secondary"
                    />
                    <Movie movie={movie} />
                </TopSection>
                <GenreMovieList genre={genre} />
            </section>
        );
    }
}

const mapStateToProps = state => ({
    query: state.query,
    searchBy: state.searchBy,
    sortBy: state.sortBy,
    movie: state.movie,
});

const mapDispatchToProps = { getMovie, showMovie };

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(MoviePage);
