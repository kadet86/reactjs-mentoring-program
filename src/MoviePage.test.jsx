import React from 'react';
import {shallow} from 'enzyme';
import MoviePage from './MoviePage';

describe('MoviePage component', () => {
    it('has persistent default snapshot', () => {
        const component = shallow(<MoviePage />);
        expect(component).toMatchSnapshot();
    });

    it('has persistent snapshot with movie={{genres: []}}', () => {
        const component = shallow(<MoviePage movie={{genres: []}}/>);
        expect(component).toMatchSnapshot();
    });

    it('has persistent snapshot with movie={{genres: ["action"]}}', () => {
        const component = shallow(<MoviePage movie={{genres: ["action"]}}/>);
        expect(component).toMatchSnapshot();
    });
});
