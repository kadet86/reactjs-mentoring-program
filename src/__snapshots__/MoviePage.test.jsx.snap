// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviePage component has persistent default snapshot 1`] = `ShallowWrapper {}`;

exports[`MoviePage component has persistent snapshot with movie={{genres: ["action"]}} 1`] = `ShallowWrapper {}`;

exports[`MoviePage component has persistent snapshot with movie={{genres: []}} 1`] = `ShallowWrapper {}`;
